astropy/convolution/convolve.py

```diff
@@ -670,20 +686,37 @@ def convolve_fft(array, kernel, boundary='fill', fill_value=0.,
     if fft_pad:  # default=True
         if psf_pad:  # default=False
             # add the dimensions and then take the max (bigger)
-            fsize = 2 ** np.ceil(np.log2(
-                np.max(np.array(arrayshape) + np.array(kernshape))))
+            bsize = np.max(np.array(arrayshape) + np.array(kernshape))
+
+            # Extra possible zero pad for dealias
+            if dealias:
+                bsize += np.ceil(bsize / 2)
+
+            fsize = 2 ** np.ceil(np.log2(np.max(bsize)))
```

Comment:
```suggestion
            fsize = 2 ** np.ceil(np.log2(bsize))
```
`bsize` should already be a single scalar here as on L705, right?